evalml/pipelines/components/transformers/preprocessing/polynomial_detrender.py

```diff
@@ -0,0 +1,112 @@
+import pandas as pd
+import woodwork as ww
+from skopt.space import Integer
+
+from evalml.pipelines.components.transformers.transformer import Transformer
+from evalml.utils import (
+    _convert_woodwork_types_wrapper,
+    import_or_raise,
+    infer_feature_types
+)
+
+
+class PolynomialDetrender(Transformer):
+    """Removes trends from time series by fitting a polynomial to the data."""
+    name = "Polynomial Detrender"
+
+    hyperparameter_ranges = {"degree": Integer(1, 3)}
+
+    def __init__(self, degree=1, random_seed=0, **kwargs):
+        """Initialize the PolynomialDetrender.
+
+        Arguments:
+            degree (int): Degree for the polynomial. If 1, linear model is fit to the data.
+                If 2, quadratic model is fit, etc. Default of 1.
+            random_seed (int): Seed for the random number generator. Defaults to 0.
+        """
+        if not isinstance(degree, int):
+            if isinstance(degree, float) and degree.is_integer():
+                degree = int(degree)
+            else:
+                raise TypeError(f"Parameter Degree must be an integer!: Received {type(degree).__name__}")
+
+        params = {"degree": degree}
+        params.update(kwargs)
+        error_msg = "sktime is not installed. Please install using 'pip install sktime'"
+
+        trend = import_or_raise("sktime.forecasting.trend", error_msg=error_msg)
+        detrend = import_or_raise("sktime.transformations.series.detrend", error_msg=error_msg)
+
+        detrender = detrend.Detrender(trend.PolynomialTrendForecaster(degree=degree))
+
+        super().__init__(parameters=params,
+                         component_obj=detrender,
```

Comment:
Is there a reason we need to set `component_obj` for this component? It appears you're doing custom logic for `fit`/`transform`/`fit_transform`, and that was the main utility of passing `component_obj` down to the base class